packages/runtime/container-runtime/src/gc/gcTelemetry.ts

```diff
@@ -186,27 +181,49 @@ export class GCTelemetryTracker {
 					return undefined;
 			}
 		})();
-		const { persistedGcFeatureMatrix, ...configs } = this.configs;
-		const unrefEventProps = {
-			trackedId,
-			type: nodeType,
-			unrefTime: nodeStateTracker?.unreferencedTimestampMs ?? -1,
-			age:
-				nodeStateTracker !== undefined
-					? currentReferenceTimestampMs - nodeStateTracker.unreferencedTimestampMs
-					: -1,
-			timeout,
-			isTombstoned,
+
+		const loggingProps: ITelemetryPropertiesExt = {
 			...tagCodeArtifacts({ id: untaggedId, fromId: untaggedFromId }),
-			...otherNodeUsageProps,
-			...this.createContainerMetadata,
+			...tagCodeArtifacts({ pkg: packagePath?.join("/") }),
+			...tagCodeArtifacts({ fromPkg: fromPackagePath?.join("/") }),
 			gcConfigs: { ...configs, ...persistedGcFeatureMatrix },
-		} satisfies Omit<IUnreferencedEventProps, "state" | "usageType"> &
-			typeof otherNodeUsageProps;
+			details: {
+				age,
+				unrefTime,
+				timeout,
+				type: nodeType,
+				trackedId,
+				isTombstoned,
+				...headers,
+				...additionalProps,
+				...otherNodeUsageProps,
+				...this.createContainerMetadata,
+			},
+		};
 
 		// If the node that is used is tombstoned, log a tombstone telemetry.
 		if (isTombstoned) {
-			this.logTombstoneUsageTelemetry(unrefEventProps, nodeType, usageType, packagePath);
+			this.logTombstoneUsageTelemetry(
+				nodeType,
+				usageType,
+				loggingProps,
+				headers?.allowTombstone,
+			);
+		}
+
+		// If the node has been referenced before it was used, this usage is not unexpected. Check the trackedId
+		// as well for cases where a DDS is used after it's data store is referenced.
+		if (
+			this.referencesSinceLastRun.has(untaggedId) ||
+			this.referencesSinceLastRun.has(trackedId)
```

Comment:
Feels like a bummer to have to account for this here. Feels like the DDS story is really shaky. I guess that's probably true because it's halfway supported.

If a DDS is used after its data store is referenced, the DDS is known to be referenced because of the implicit references from DataStore to DDS.

Let's think of the general case -- What if A -> B (both DataStores) but A is unreferenced, then outbound route is added to A, now B is referenced too. Do we handle that general case?  Can we?

If not, then I guess this is fine.  If we can handle that general case, then this isn't needed.

Reply:
Good point. This does change the behavior. Before this change, after GC runs, B will be referenced and we will not log if it was used. With this change, we will log for B.

I don't think there is a straightforward way to handle this case. We could, but it will be more complicated than it already is. So, I think this change may not be worth it.